File: tests/fixtures/redis-py-app/stream.py
import json

from client import r


def append_event(event: dict) -> str:
    return r.xadd("events-stream", {"data": json.dumps(event)})


def read_events(last_id: str = "0") -> list:
    return r.xread({"events-stream": last_id}, block=0)

File: tests/fixtures/ioredis-app/src/stream.ts
// @ts-nocheck
import { redis } from './client';

export async function appendEvent(event: unknown) {
  return redis.xadd('events-stream', '*', 'data', JSON.stringify(event));
}

export async function readEvents(lastId = '0') {
  return redis.xread('BLOCK', 0, 'STREAMS', 'events-stream', lastId);
}
